File: PathStructureClient/src/dtos/JsonRpcNotification.js
/**
 * @template TParams
 */
class JsonRpcNotification {
  /**
   * @param {object} options
   * @param {string} options.method
   * @param {TParams} options.params
   * @param {string} [options.jsonrpc]
   */
  constructor({ method, params, jsonrpc = '2.0' }) {
    /** @type {string} */
    this.jsonrpc = jsonrpc;
    /** @type {string} */
    this.method = method;
    /** @type {TParams} */
    this.params = params;
  }
}

module.exports = { JsonRpcNotification };

File: PathStructure.WatcherHost/Dtos/JsonRpcNotification.cs
using System;

namespace PathStructure.WatcherHost.Dtos
{
    /// <summary>
    /// Represents a JSON-RPC notification payload.
    /// </summary>
    /// <typeparam name="TParams">The parameter payload type.</typeparam>
    internal sealed class JsonRpcNotification<TParams>
    {
        /// <summary>
        /// Gets or sets the JSON-RPC protocol version.
        /// </summary>
        public string Jsonrpc { get; set; } = "2.0";

        /// <summary>
        /// Gets or sets the notification method name.
        /// </summary>
        public string Method { get; set; }

        /// <summary>
        /// Gets or sets the notification parameters.
        /// </summary>
        public TParams Params { get; set; }
    }
}
